Remotion/Data/DomainObjects.UnitTests/IntegrationTests/Transaction/ClientTransactionCommitValidationTest.cs

```diff
@@ -28,9 +28,7 @@ public class ClientTransactionCommitValidationTest : ClientTransactionBaseTest
     [Test]
```

Comment:
Fun fact: Using `MinActiveRowVersion` does not work in a transaction. Why do we need to verify that nothing is written to the database if we are using an umbrella transaction anyway? Couldn't we just make sure that the umbrella transaction hasn't commited?

Reply:
@a-ctor 
> Using MinActiveRowVersion does not work in a transaction. 
Please elaborate because I have code that uses 
```
provider.Connect();
var command = provider.CreateDbCommand();
command.CommandText = "SELECT CONVERT (BIGINT, MIN_ACTIVE_ROWVERSION()) -1";
object maxTimestamp = command.ExecuteScalar();
```
inside a System.Transaction scope.

As for the need: we need a check that will fail the test if a database write is detected because the test must verify that no data is written to the database. Up until now, this was done implicityly because the database was readonly. An alternative approach would be to inject some code into the infrastrucutre that will throw an exception if StorageProvider.Save gets called, which is a much more involved operation.

Reply:
I changed a test to insert an item and used the section around the commit:

```
var newObject = CreateClassWithAllDataTypes();
newObject.ByteProperty = 13;
newObject.StringWithNullValueProperty = null;
newObject.NullableBinaryProperty = null;

using (var s = OpenNoDatabaseWriteSection())
{
  Assert.That(() => TestableClientTransaction.Commit(), Throws.Nothing);
}
```

This code should fail even if we are in a transaction, right? But it does not. If I remove the umbrella transaction in the `DatabaseTest` then this test fails because something has been written to the DB and the `MIN_ACTIVE_ROWVERSION` changed. Maybe that is expected? I mean we are trying to ensure that nothing is written and it isn't but that was known from the start since we have an umbrella transaction...

Reply:
Use @@DBTS instead